src/request.ts

```diff
@@ -0,0 +1,45 @@
+import { readFileSync } from "fs";
+import { Result, Ok, Err } from "./result";
+
+/**
+ * An object that represents an error with a request
+ * @public
+ */
+export interface RequestError {
+  message: string;
+  statusCode?: number;
+}
+
+export async function doFetch({
+  urlPath,
+  ...rest
+}: {
+  urlPath: string;
+} & RequestInit): Promise<Result<Response, RequestError>> {
+  const response = await fetch(new URL(urlPath), rest);
```

Comment:
Do you wanna wrap this in try too? Fetch has this stupid API where it throws if it can't connect